src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -184,12 +193,17 @@ private DetailAST getEventAst(TreeWalkerAuditEvent event) {
         DetailAST curNode = ast;
         while (curNode != null) {
             DetailAST toVisit = curNode.getFirstChild();
-            if (toVisit != null) {
-                childAstLineNoSet.add(toVisit.getLineNo());
-            }
+            addChildAstLineNo(childAstLineNoSet, curNode);
+            addChildAstLineNo(childAstLineNoSet, toVisit);
             while (curNode != null && toVisit == null) {
                 toVisit = curNode.getNextSibling();
                 curNode = curNode.getParent();
+                if (curNode != null && curNode.equals(ast.getParent())) {
+                    break;
+                }
+            }
+            if (curNode != null && curNode.equals(ast.getParent())) {
+                break;
```

Comment:
same complaint from https://github.com/checkstyle/patch-filters/pull/137#discussion_r460453311.
you can move these to `while` condition

Reply:
done, but the second `if (curNode != null && curNode.equals(ast.getParent())) {` I do not change, because the next line is `curNode = toVisit;`, we can not move logic to `main` while condition